pages/[walletAddress]/index.tsx

```diff
@@ -0,0 +1,21 @@
+import { validateAddress, ValidationResult } from "@taquito/utils";
+import { GetServerSideProps } from "next";
+
+export const getServerSideProps: GetServerSideProps = async ({ params }) => {
+  if (!params || !params.walletAddress || Array.isArray(params.walletAddress))
+    return { notFound: true };
+
+  if (validateAddress(params.walletAddress) !== ValidationResult.VALID)
+    return { notFound: true };
+
+  return {
+    redirect: {
+      destination: `/${params.walletAddress}/proposals`,
+      permanent: true,
+    },
+  };
+};
+
+export default function WalletAddressIndex() {
```

Comment:
what's this function for?

Reply:
I have no choice than exporting a component function as it's a page. And as code is only meant to run server side to redirect, we don't care about returning a content as user is redirected on the server level, so before the page load